src/main/java/hng_java_boilerplate/organisation/service/OrganisationService.java

```diff
@@ -0,0 +1,80 @@
+package hng_java_boilerplate.organisation.service;
+
+import hng_java_boilerplate.organisation.dto.CreateOrganisationRequestDto;
+import hng_java_boilerplate.organisation.dto.CreateOrganisationResponseDto;
+import hng_java_boilerplate.organisation.dto.DataDto;
+import hng_java_boilerplate.organisation.entity.Organisation;
+import hng_java_boilerplate.organisation.exception.OrganisationNameAlreadyExistsException;
+import hng_java_boilerplate.organisation.repository.OrganisationRepository;
+import hng_java_boilerplate.user.entity.User;
+import hng_java_boilerplate.user.repository.UserRepository;
+import lombok.RequiredArgsConstructor;
+import org.springframework.security.core.Authentication;
+import org.springframework.stereotype.Service;
+
+import java.time.LocalDateTime;
+import java.util.List;
+
+@Service
+@RequiredArgsConstructor
+public class OrganisationService {
+    private final OrganisationRepository organisationRepository;
+    private final UserRepository userRepository;
+
+    public CreateOrganisationResponseDto create(
+            CreateOrganisationRequestDto orgRequest,
```

Comment:
why not do this using transaction to ensure atomicity.

Reply:
made create organisation transactional.